core/src/main/scala/org/apache/spark/deploy/Client.scala

```diff
@@ -67,8 +67,15 @@ private class ClientEndpoint(
         //       people call `addJar` assuming the jar is in the same directory.
         val mainClass = "org.apache.spark.deploy.worker.DriverWrapper"
 
-        val classPathConf = "spark.driver.extraClassPath"
-        val classPathEntries = sys.props.get(classPathConf).toSeq.flatMap { cp =>
+        val driverClassPathConf = "spark.driver.extraClassPath"
+        val commonClassPathConf = "spark.common.extraClassPath"
+
+        val driverClassPath = Some(
+          (sys.props.get(commonClassPathConf) ++ sys.props.get(driverClassPathConf))
```

Comment:
Not your fault; but it's slightly weird that this code is using `sys.props` instead of the `SparkConf` object provided in the constructor.